workflow/rules/annotate.smk

```diff
@@ -1,101 +1,117 @@
+        gene_annotation="resources/gene_annotation.gff3.gz",
+        regulatory_annotation="resources/regulatory_annotation.gff3.gz",
+        repeat_annotation=lambda wc: "resources/repeat_masker.fa.out.gz"
+        if config["reference"].get("repeat_masker_download_link", "")
+        else "",
     output:
-        variants="results/calling/calls/annotated/{sample}.bcf",
-        annotation=temp("results/calling/candidates/{sample}.sorted.bcf.tab"),
-        annotation_bgzip=temp("results/calling/candidates/{sample}.sorted.bcf.tab.bgz"),
-        annotation_bgzip_tabix=temp(
-            "results/calling/candidates/{sample}.sorted.bcf.tab.bgz.tbi"
-        ),
+        annotated="results/calling/graphs/{group}.annotated.graph",
+    threads: 1
     log:
-        "logs/re-annotate/{sample}.log",
+        "logs/cyrcular_annotate_graph/{group}.log",
     benchmark:
-        "benchmarks/re-annotate/{sample}.txt"
+        "benchmarks/cyrcular_annotate_graph/{group}.txt"
     conda:
-        "../envs/vcf_annotate.yaml"
+        "../envs/cyrcular.yaml"
     params:
-        header=copy_annotation_vembrane_header_expr(),
-        table_expr=copy_annotation_table_expr(),
-        columns=copy_annotation_bcftools_annotate_columns(),
+        repeat_annotation=lambda wc, input: f"  --repeat-annotation {input.repeat_annotation} "
+        if config["reference"].get("repeat_masker_download_link", "")
+        else "",
+    shell:
+        "cyrcular graph annotate "
+        "  --reference {input.reference} "
+        "  --gene-annotation {input.gene_annotation} "
+        "  --regulatory-annotation {input.regulatory_annotation} "
+        "  --output {output.annotated} "
+        "  {input.graph} "
+        "2> {log} "
+
+
+rule reheader_filtered_bcf:
+    input:
+        bcf="results/calling/calls/filtered_fdr/{group}.bcf",
+        sorted_header="results/calling/calls/filtered_fdr/reheader/{group}.header.sorted.txt",
+    output:
+        bcf="results/calling/calls/filtered_fdr/reheader/{group}.bcf",
+    log:
+        "logs/reheader_filtered_bcf/{group}.log",
+    conda:
+        "../envs/bcftools.yaml"
     shell:
+        ## bcftools re-header seems to re-order entries
+        # bcftools reheader --header {input.sorted_header} --output {output.bcf} {input.bcf}
+        ## so we have to re-header ourselves
         """
-        vembrane table --header {params.header:q} {params.table_expr:q} {input.candidates_with_annotation} > {output.annotation} 2> {log}
-        bgzip -c {output.annotation} > {output.annotation_bgzip} 2>> {log}
-        tabix -p vcf --zero-based -S 1 -f {output.annotation_bgzip} 2>> {log}
-        bcftools annotate --header-lines {input.header_lines} --annotations {output.annotation_bgzip} --columns {params.columns} --output-type b --output {output.variants} {input.variants}  2>> {log}
+        cat {input.sorted_header} <(bcftools view -H {input.bcf}) | bcftools view -Ob > {output.bcf} 2> {log}
```

Comment:
The manual re-header of bcf files works for now, but is an ugly workaround. Might need some investigating in the future.

Reply:
So, should we add a `TODO`, here?